logical/logical.go

```diff
@@ -2,6 +2,32 @@ package logical
 
 import log "github.com/mgutz/logxi/v1"
 
+// BackendType is the type of backend that is being implemented
+type BackendType uint32
+
+// The these are the types of backends that can be derived from
+// logical.Backend
+const (
+	TypeUnknown     BackendType = 0 // This is also the zero-value for BackendType
+	TypeLogical     BackendType = 1
+	TypeCredential  BackendType = 2
+	TypePassthrough BackendType = 3
```

Comment:
This actually brings up a couple of good points:

1) Will we need to have other backends in the future whose leases are ignored (unlikely), and if so, does the name `TypePassthrough` make sense to describe them (no, since passthrough is the actual name of the backend).

2) Is it actually that much faster to check this than a type assertion. I don't know, since even if the value being checked at runtime may be one of several possible values, in theory the compiler can know whether any such value implements the interface at compile time.

Item number 1 above makes me think that the right approach isn't to special case the type, but rather to add a set of flags, of which right now (and in the near future) there is only one (IgnoreLeases).